server/src/main/java/org/opensearch/cluster/routing/remote/RemoteRoutingTableService.java

```diff
@@ -0,0 +1,223 @@
+            assert uploadedIndexMetadataOptional.isPresent() == true;
+            return uploadedIndexMetadataOptional.get();
+        }).collect(Collectors.toList());
+    }
+
+    public static List<String> getIndicesRoutingDeleted(RoutingTable previousRoutingTable, RoutingTable currentRoutingTable) {
+        List<String> deletedIndicesRouting = new ArrayList<>();
+        for(IndexRoutingTable previousIndexRouting: previousRoutingTable.getIndicesRouting().values()) {
+            if(!currentRoutingTable.getIndicesRouting().containsKey(previousIndexRouting.getIndex().getName())) {
+                // Latest Routing Table does not have entry for the index which means the index is deleted
+                deletedIndicesRouting.add(previousIndexRouting.getIndex().getName());
+            }
+        }
+        return deletedIndicesRouting;
+    }
+
+    public static List<String> getIndicesRoutingUpdated(RoutingTable previousRoutingTable, RoutingTable currentRoutingTable) {
+        List<String> updatedIndicesRouting = new ArrayList<>();
+        for(IndexRoutingTable currentIndicesRouting: currentRoutingTable.getIndicesRouting().values()) {
+            if(!previousRoutingTable.getIndicesRouting().containsKey(currentIndicesRouting.getIndex().getName())) {
+                // Latest Routing Table does not have entry for the index which means the index is created
+                updatedIndicesRouting.add(currentIndicesRouting.getIndex().getName());
+            } else {
+                if(previousRoutingTable.getIndicesRouting().get(currentIndicesRouting.getIndex().getName()).equals(currentIndicesRouting)) {
+                    // if the latest routing table has the same routing table as the previous routing table, then the index is not updated
+                    continue;
+                }
+                updatedIndicesRouting.add(currentIndicesRouting.getIndex().getName());
+            }
+        }
+        return updatedIndicesRouting;
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (blobStoreRepository != null) {
+            IOUtils.close(blobStoreRepository);
+        }
+    }
+
+    public void start() {
+        assert isRemoteRoutingTableEnabled(settings) == true : "Remote routing table is not enabled";
+        final String remoteStoreRepo = settings.get(
+            Node.NODE_ATTRIBUTES.getKey() + RemoteStoreNodeAttribute.REMOTE_STORE_CLUSTER_STATE_REPOSITORY_NAME_ATTRIBUTE_KEY
+        );
+        assert remoteStoreRepo != null : "Remote routing table repository is not configured";
+        final Repository repository = repositoriesService.get().repository(remoteStoreRepo);
+        assert repository instanceof BlobStoreRepository : "Repository should be instance of BlobStoreRepository";
+        blobStoreRepository = (BlobStoreRepository) repository;
+    }
+
+    public static class RemoteIndexRoutingResult {
```

Comment:
Why is this class needed? `IndexRoutingTable` already holds `Index` info

Reply:
The idea was to keep something similar to `RemoteResult` type of object. For now, I have removed it. We can take an call when the parent PRs are merged.